core/src/cluster_info.rs

```diff
@@ -3100,31 +2947,28 @@ impl ClusterInfo {
 /// Returns Neighbor Nodes and Children Nodes `(neighbors, children)` for a given node based on its stake
 pub fn compute_retransmit_peers(
     fanout: usize,
-    my_index: usize,
-    stakes_and_index: Vec<usize>,
-) -> (Vec<usize>, Vec<usize>) {
-    //calc num_layers and num_neighborhoods using the total number of nodes
-    let (num_layers, layer_indices) =
-        ClusterInfo::describe_data_plane(stakes_and_index.len(), fanout);
-
-    if num_layers <= 1 {
-        /* single layer data plane */
-        (stakes_and_index, vec![])
-    } else {
-        //find my layer
-        let locality = ClusterInfo::localize(&layer_indices, fanout, my_index);
-        let upper_bound = cmp::min(locality.neighbor_bounds.1, stakes_and_index.len());
-        let neighbors = stakes_and_index[locality.neighbor_bounds.0..upper_bound].to_vec();
-        let mut children = Vec::new();
-        for ix in locality.next_layer_peers {
-            if let Some(peer) = stakes_and_index.get(ix) {
-                children.push(*peer);
-                continue;
-            }
-            break;
-        }
-        (neighbors, children)
-    }
+    node: usize,
+    index: &[usize],
+) -> (Vec<usize> /*neighbors*/, Vec<usize> /*children*/) {
+    // 1st layer: fanout    nodes starting at 0
+    // 2nd layer: fanout**2 nodes starting at fanout
+    // 3rd layer: fanout**3 nodes starting at fanout + fanout**2
+    // ...
+    // Each layer is divided into neighborhoods of fanout nodes each.
+    let offset = node % fanout; // Node's index within its neighborhood.
+    let anchor = node - offset; // First node in the neighborhood.
+    let neighbors = (anchor..)
+        .take(fanout)
+        .map(|i| index.get(i).copied())
+        .while_some()
+        .collect();
+    let children = ((anchor + 1) * fanout + offset..)
+        .step_by(fanout)
+        .take(fanout)
+        .map(|i| index.get(i).copied())
+        .while_some()
+        .collect();
+    (neighbors, children)
```

Comment:
@behzadnouri I don't know much about this turbine code. But, This new implementation looks very succinct and beautiful, considering this replaces around ~200 lines of code. Good kudos to you. :)

Reply:
Hey @ryoqun, thanks for your encouraging words : )